src/suggestion.rs

```diff
@@ -50,6 +92,151 @@ impl fmt::Display for Detector {
+pub fn convert_long_statements_to_short(
+    terminal_size: usize,
+    indent: usize,
+    stripped_line: &str,
+    offset: &mut usize,
+    range_word: Range,
+    padding_till_literal_start: usize,
+    marker_size: &mut usize,
+) -> String {
+    use super::*;
+    //
+    // The paddings give some space for the ` {} ...` and extra indentation and formatting:
+    //
+    //|---------------------------------------------------------------------------------------| terminal_size
+    //|-------| padding_till_literal_start = indent (3+line_number_digit_count) + 2 white spaces = 7usize, for this case.
+    //     |------| offset = PADDING_OFFSET; 3 chars for `...` and 2 white spaces more added in the formatting.
+    //
+    //   --> /home/tmhdev/Documents/cargo-spellcheck/src/suggestion.rs:62
+    //    |
+    // 62 |  ... Reasn of food, what's up with pie? There's strawberry pie, apple, pumpkin ...
+    //    |      ^^^^^^
+    //    | - there, Cherie, thither, or tither
+    //    |
+    //    |   Possible spelling mistake found.
+    //
+
+    const PADDING_OFFSET: usize = 5;
+    const TOO_LONG_WORD: usize = 20;
+    const DISPLAYED_LONG_WORD: usize = 4;
+    const PADDING_AROUND_LONG_LINES: usize = 10;
+
+    // We will be using ranges to help doing the fitting:
+    //
+    // |----------------------------------literal_word----------------------------------|
+    // |----------------------|---------misspelled_word---------|-----------------------|
+    // |-----left_context-----|---start_word----|----end_word---|-----right_context-----|
+    //
+    // Obs: paddings are not being considered in the illustration, but info is above.
+
+    // the line being analysed can affect how the indentation is done
+    // this values is dynamically calculated according to the line number
+    let mut range_start_word = Range {
+        start: range_word.start,
+        end: range_word.start,
+    };
+    let mut range_end_word = Range {
+        start: range_word.end,
+        end: range_word.end,
+    };
+    let mut misspelled_word: String = stripped_line
+        .chars()
+        .skip(range_word.start)
+        .take(range_word.len())
+        .collect();
+    // Check words that are considered too long; Word will be formatted for fitting
+    if range_word.len() > TOO_LONG_WORD {
+        range_start_word = Range {
+            start: range_word.start,
+            end: range_start_word.start + DISPLAYED_LONG_WORD,
+        };
+        range_end_word = Range {
+            start: range_word
+                .end //non inclusive
+                .saturating_sub(DISPLAYED_LONG_WORD),
+            end: range_word.end,
+        };
+
+        //  too long word will be shorter as it follows:
+        //    4 chars |----|  ... |---| 3 chars
+        //                ther...eee, for therieeeeeeeeeeeeeeee
+        //
+        misspelled_word = format!(
+            "{}...{}",
+            stripped_line
```

Comment:
create intermediate variables for these and then...

Reply:
```
        let misspelled_word_start = stripped_line.chars()
                .skip(range_start_word.start)
                .take(range_start_word.len())
                .collect::<String>();
        let misspelled_word_end = stripped_line.chars()
                .skip(range_start_word.end.saturating_sub(3))
                .take(3)
                .collect::<String>();
        misspelled_word = format!(
            "{}...{}",
            misspelled_word_start,
            misspelled_word_end
        );
        *marker_size = misspelled_word_end.chars().count() + misspelled_word_end.chars().count();

```
I think you meant this to avoid going over the chars many times.
```
 let stripped_line_chars = stripped_line.chars();
        misspelled_word = format!(
            "{}...{}",
            stripped_line_chars
                .skip(range_start_word.start)
                .take(range_start_word.len())
                .collect::<String>(),
            stripped_line_chars
                .skip(range_start_word.end.saturating_sub(3))
                .take(3)
                .collect::<String>()
        );

        *marker_size = stripped_line_chars.count();
```

Tried using references/clone as well. I had problems with this:
`[Note] move occurs because `stripped_line_chars` has type `std::str::Chars<'_>`, which does not implement the `Copy` trait`

Reply:
I meant to make a temp var for each of the two components.

What you are attempting will not work, since iterators (here `.chars()` ) are lazily executed. So the iterator type (or accessor struct) will be consumed by `collect()` https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.collect

Reply:
cool, then it is okay!